### PR TITLE
Update bound_iam_role_arn to make it work for Vault provider >= 1.1.4

### PR DESCRIPTION
**Pull request by guangie88**
Also add reminder in README.md

### CODE REVIEW

File: modules/aws-auth/README.md [hunk 1]
```diff
@@ -27,6 +27,11 @@ initialised and unsealed the Vault servers.
 
 ## Vault Provider
 
+You must be using Vault provider version `>= 1.1.4` due to the breaking changes in many of the field
```
Comment (lawliet89) [inline modules/aws-auth/README.md]: `~> 1.1.4` is better.

Comment (guangie88) [reply modules/aws-auth/README.md]: Okay

Comment (guangie88) [reply modules/aws-auth/README.md]: Done

**Comment by guangie88** [comment]
Lol sorry forgot to push the version syntax change, please see https://github.com/GovTechSG/terraform-modules/pull/167.

**Pull request merged**